### DISCUSSION TITLE
Little nudge to start for the solution

### DISCUSSION THREAD

**Discussion by SuryaViswanath11** [pymupdf/PyMuPDF#957] [Looking for help]
I have a pdf file and I am trying to annotate over the different placeholders. I wanted to know if there is a way that I can annotate text data over these multiple boxes on the pdf. 
Here is the pdf file that I am trying to work on.

[Form_49A.PDF](https://github.com/pymupdf/PyMuPDF/files/6161461/Form_49A.PDF)

thanks and regards
Surya

**Comment by JorjMcKie** [comment]
> annotate text data over these multiple boxes

What does that mean? Help info for how to fill out the fields?

**Reply by SuryaViswanath11** [reply]
so in the pdf file, there are many boxes present. We need to fill out those boxes with the respective information 

for ex: there is a field called surname, we have the surname as say: DOUGLAS, we need to fill them in the individual boxes present in the pdf file and annotate the complete PDF

**Reply by JorjMcKie** [reply]
Well, I **_think_** you refer to the field label. This `Widget` attribute contains text that shows up when the cursor hovers over a field. Fill it like this: `widget.field_label = "arbitrary text - e.g. to help filling out the field"`.

**Reply by JorjMcKie** [reply]
like in this example:
[widgettest.pdf](https://github.com/pymupdf/PyMuPDF/files/6170175/widgettest.pdf)

**Reply by SuryaViswanath11** [reply]
But the pdf file that I have attached, the textboxes present doesn't act like widgets. Will this method work on that

**Reply by JorjMcKie** [reply]
Of course not - this only works for widgets / fields.
If you want to embed explanations / help in the general case, the only way I can think of is inserting 'Text' annotations. They are represented by little icons, which show stored text, when the cursors hovers over it - see method `page.addTextAnnot()` documentation.
Icons can be chosen from this list:
![mupdf-icons](https://user-images.githubusercontent.com/8290722/112322294-2722d580-8c87-11eb-937e-91d8d58f2554.jpg)

**Reply by SuryaViswanath11** [reply]
is there a way to extract shapes, and identify its location to annotate? or is there a way to hardcode the coordinates for each boxes to fill in the details?

**Reply by JorjMcKie** [reply]
When you say "shape" you mean things like lines and rectangles? This is done by `page.get_drawings()`.
What do you mean by "hardcode"? You can of course do `page.addTextAnnot(point, ....)` with the same value for `point` on every page.

**Reply by SuryaViswanath11** [reply]
since the document doesn't change, I was wondering if I can get the annotations of those rectangles and using that information I was hoping to use these values to annotate using those coordinates.
